src/reducers/game.ts

```diff
@@ -161,6 +170,45 @@ const gameSlice = createSlice({
             return state
         },
 
+        createEffect(state, action: PayloadAction<{ newEffect: string }>): GameState {
+            const newEffectName = action.payload.newEffect
+            const newEffectID = generateEffectID(newEffectName)
+
+            if (newEffectID in state.availableEffects) {
+                alert(`Effect "${newEffectName}" with ID "${newEffectID}" already exists`)
+            } else {
+                state.availableEffects[newEffectID] = newEffectName
+            }
+
+            return state
+        },
+
+        deleteEffect(state, action: PayloadAction<string>): GameState {
+            const effectID = action.payload
+            state.players.forEach(player => {
+                const effects = player.effects.filter(effect => effect !== effectID)
+                player.effects = [...effects]
+            })
+            delete state.availableEffects[effectID]
+            return state
+        },
+
+        togglePlayerEffect(state, action: PayloadAction<{ playerID: number, effectID: string, disable?: ("on"|"off") }>): GameState {
```

Comment:
```suggestion
        togglePlayerEffect(state, action: PayloadAction<{ playerID: number, effectID: string }>): GameState {
```

remove this flag logic again.